test/support/src/helpers.h

```diff
@@ -759,6 +759,25 @@ std::string get_fragment_dir(std::string array_dir);
  */
 std::string get_commit_dir(std::string array_dir);
 
+/* List the given path and store the counts of each file type. */
+class CommitsDirectory {
```

Comment:
Now that I see this implementation, I'm thinking of something that we could use for everything... We could make a FileCount base class that takes in a vector of expected extensions at construction:
std::vector<std::string> expected_extensions,
This would be used at construction to make sure that every file in the directory has a file extension we expect as we count the files:
auto files = vfs.ls(path);
for (auto file : files) {
  auto file_ext = /* parse file extension */
  if (std::find(expected_extensions.begin(), expected_extensions.end(), file_ext) == vec.end()) {
    throw exception...
  }
  file_count_[ext]++;
}

The CommitsDirectory constructor becomes something like:
CommitsDirectory() : FileCount({tiledb::sm::constants::vacuum_file_suffix, tiledb::sm::constants::write_file_suffix... ... ...})

That was we could create other classes for other directories, ie: GroupDirectory... Does that make sense?

Let me know if you are under a time constraint to implement this and we can do this in a follow up PR.